common/src/main/java/noobanidus/mods/lootr/common/block/entity/BlockEntityTicker.java

```diff
@@ -167,44 +159,29 @@ public static void onServerTick() {
         if (LootrAPI.resolveBlockEntity(newBlockEntity) instanceof ILootrBlockEntity && newBlockEntity instanceof RandomizableContainerBlockEntity rbe) {
           rbe.setLootTable(table, seed);
         } else {
-          LootrAPI.LOG.error("replacement " + replacement + " is not an ILootrBlockEntity " + entry.getDimension() + " at " + entry.getPosition());
+          LootrAPI.LOG.error("replacement {} is not an ILootrBlockEntity {} at {}", replacement, entry.getDimension(), entry.getPosition());
         }
 
-        toRemove.add(entry);
+        iterator.remove();
       }
     }
     synchronized (listLock) {
-      tickingList = true;
-      blockEntityEntries.removeAll(toRemove);
       blockEntityEntries.addAll(pendingEntries);
-      tickingList = false;
       pendingEntries.clear();
```

Comment:
As mentioned earlier, it might be worth it to use a `ConcurrentHashMap` for `pendingEntries` instead.